v2/pkg/engine/resolve/fetch.go

```diff
@@ -429,12 +370,28 @@ type FetchReason struct {
 	IsRequires  bool     `json:"is_requires,omitempty"`
 }
 
+// FetchInfo contains additional (derived) information about the fetch.
+// Some fields may not be generated depending on planner flags.
 type FetchInfo struct {
 	DataSourceID   string
 	DataSourceName string
 	RootFields     []GraphCoordinate
 	OperationType  ast.OperationType
 	QueryPlan      *QueryPlan
+
+	// CoordinateDependencies contain a list of GraphCoordinates (typeName+fieldName)
+	// and which fields from other fetches they depend on.
+	// This information is useful to understand why a fetch depends on other fetches,
+	// and how multiple dependencies lead to a chain of fetches
+	CoordinateDependencies []FetchDependency
+
+	// FetchReasons contains provenance for reasons why particular fields were fetched.
+	// If this structure is built, then all the fields are processed.
+	FetchReasons []FetchReason
+
+	// PropagatedFetchReasons holds those FetchReasons that will be propagated
```

Comment:
Could we extend a comment that this will contain information only for the fields which was marked as requiresFetchReason stored in ds configuration